### DISCUSSION TITLE
Como testar reCaptcha?

### DISCUSSION THREAD

**Discussion by didiraja** [frontendbr/forum#736] [Perguntas e Respostas]
Galera, ultimamente os clientes estavam reclamando porque recebiam muitos leads ou vazios ou de spam, daí pediram para eu implementar o reCaptcha v2. Até aí tudo certo, achei tutorial no youtube e fiz a validação no back-end bonitinha, mas como eu testar que realmente está funcionando?

Nos jobs internos, eu sei que o captcha resolveu porque o pessoal de mídia mostrou os relatórios, mas agora me apareceu um freela e embora eu saiba com implementar, não sei como testar e verificar que ele está funcionando completamente. Alguma indicação?

P.S. A própria ferramenta tem uma parte de relatório mas nunca funcionou. Sempre diz que não teve acesso suficiente para exibir algo, mas nós aqui verificávamos internamente que estava funcionando.

**Comment by mumumilk** [comment]
Não entendi muito bem sua pergunta @didiraja , poderia ser mais específico na parte de testar o reCaptcha?

**Comment by felipefialho** [comment]
Também não entendi...

**Comment by lucasmahle** [comment]
Você se refere em testar a credibilidade do reCaptcha, no sentido de se realmente ele funciona?
Ou você está se referindo em testar se está aparecendo pro usuário interagir?

**Comment by didiraja** [comment]
@lucasmashlrle isso testar se a ferramenta está realmente barrando robôs

**Comment by mumumilk** [comment]
Não sei como está sendo com você, mas os gráficos que a google fornece estão batendo por aqui. Quando eu implementei, realmente apareceu a mensagem que não havia acessos suficientes, mas depois que a galera começou a acessar os gráficos foram montados.

Detalhe que, se você estiver usando a key disponivel para testes automatizados (a que não faz validação), daí realmente não vai aparecer, creio eu.

![image](https://user-images.githubusercontent.com/21252014/31184620-45816762-a900-11e7-9381-b463ce753f12.png)
